repository_service_tuf_worker/repository.py

```diff
@@ -1121,3 +1202,96 @@ def metadata_rotation(
         logging.warn(deprecation_message)
 
         return self.metadata_update(payload, update_state)
+
+    def _sign_root(self, root: Metadata[Root]) -> Tuple[bool, str]:
+        """
+        Sign the root metadata and call actions depending on the bootstrap
+        state, and if it matches the threshold
+        """
+        signed = self._validate_signatures(root, Root.type)
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+
+        if signed and "signing" not in bootstrap:
+            self._root_metadata_update(root)
+            message = f"Root version {root.signed.version} signed."
+
+        elif signed and "signing" in bootstrap:
+            bootstrap_task_id = bootstrap.split("signing-")[1]
+            self._persist(root, Root.type)
+            self._bootstrap_online_roles(root, bootstrap_task_id)
+            message = "Bootstrap finished"
+
+        else:
+            self.write_repository_settings(
+                f"{Root.type.upper()}_SIGNING", root.to_dict()
+            )
+            message = f"Root version {root.signed.version} pending sign."
+
+        return signed, message
+
+    def sign_metadata(
+        self,
+        payload: Dict[str, str],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        """
+        Start the signing metadata process
+        """
+        rolename = payload["role"]
+        sig_dict = payload["signature"]
+
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+        if bootstrap is None:
+            result = ResultDetails(
+                status="Task finished.",
+                details={
+                    "signed": False,
+                    "message": "Signing requires RSTUF bootstrap",
+                },
+                last_update=datetime.now(),
+            )
+
+            return asdict(result)
+
+        available_signing = list(
+            filter(lambda var: "SIGNING" in var, dir(self._settings))
+        )
+
+        if f"{rolename.upper()}_SIGNING" not in available_signing:
+            result = ResultDetails(
+                status="Task finished.",
+                details={
+                    "signed": False,
+                    "message": f"No signatures pending for {rolename}.",
+                },
+                last_update=datetime.now(),
+            )
+
+            return asdict(result)
+
+        dict_metadata = self._settings.get_fresh(f"{rolename.upper()}_SIGNING")
```

Comment:
Maybe you want to check the value of `dict_metadata`.
After `_persist` `*_SIGNING` is set to `None` and it will exist in `available_signing`, so it will execute line 1274, but `dict_metadata` won't be a dictionary.